src/cli/cli.cpp

```diff
@@ -1364,7 +1364,10 @@ void Interpreter::ProcessExtAddress(uint8_t aArgsLength, char *aArgs[])
     {
         otExtAddress extAddress;
 
-        VerifyOrExit(Hex2Bin(aArgs[0], extAddress.m8, sizeof(otExtAddress)) >= 0, error = OT_ERROR_INVALID_ARGS);
+        memset(&extAddress, 0, sizeof(extAddress));
+        VerifyOrExit(strlen(aArgs[0]) == 2 * sizeof(extAddress.m8), error = OT_ERROR_INVALID_ARGS);
```

Comment:
is this necessary?
fyi, there is length check in Hex2Bin
https://github.com/openthread/openthread/blob/72109da6f1a0bdc1ec91e0281f9ac53269a5e978/src/cli/cli.cpp#L296-L299

Reply:
`aAllowTruncate` is checking if the input string is longer than needed, but doesn't prevent shorter strings.
Also, checking the return value of `Hex2Bin` isn't sufficient as it allows odd lengths of input strings.
Is it better to change `Hex2Bin` to match exactly `hexLength == 2 * aBinLength` if `!aAllowTruncate`?

Reply:
The implementation of Hex2Bin is allowing the first preceding 0 skipped.  Do you want to forbid such usage?

Reply:
Yes. If we don't allow truncation, then truncation on neither the input string nor the output binary should be allowed.
It's confusing if the conversion is done left-to-right but allows left-most 0's. E.g. `1234567` -> `01234567` but `12345` -> `01234500`.

Reply:
`aAllowTruncate` was introduced purely for parsing a long input string with a buffer of 50 bytes in #4241 without sacrificing the flexibility of skipping first preceding 0 character.

@Moandor-y ,  when will `12345->01234500` happen? Do you mean the buffer (reset  with 0) is longer than the what the input string could be converted? If so, I think the user should check the return value of Hex2Bin to see how many bytes are converted from the input string when using the buffer.

Reply:
Updated to leave `Hex2Bin` untouched and made all callers to check the return value match exactly the binary length. This should prevent the uninitialized memory error and retain the flexibility.